lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -219,60 +220,174 @@ sub overview {
 
 =over 4
 
-=item create()
+=item _eval_param_grouping()
 
-Creates a job given a list of settings passed as parameters. TEST setting/parameter
-is mandatory and should be the name of the test.
+Removes job-specific parameters from the specified hash and returns a new hash
+containing the removed parameters (where the keys are job suffixes).
+
+So "{foo => 1, bar:jobA => 2, baz:jobB => 3}" will be turned into "{foo => 1}"
+returnig "{jobA => {bar => 2}, jobB => {baz => 3}}".
 
 =back
 
 =cut
 
-sub create {
-    my $self = shift;
-    my $params = $self->req->params->to_hash;
-    my $is_clone_job = delete $params->{is_clone_job} // 0;
+sub _eval_param_grouping ($params) {
+    my %grouped_params;
+    for my $param_key (keys %$params) {
+        my $job_suffix_start = rindex($param_key, ':');
+        next unless $job_suffix_start >= 0;
+        my $job_suffix = substr $param_key, $job_suffix_start + 1;
+        my $setting_key = substr $param_key, 0, $job_suffix_start;
+        $grouped_params{$job_suffix}->{$setting_key} = delete $params->{$param_key};
+    }
+    return \%grouped_params;
```

Comment:
Something like this maybe

```suggestion
    map { ($job, $setting) = split /:/; $setting => {$job => $x{$_}} } grep { /:/ } keys %$params;
```

For a test:

```
perl -e '%x = ("foo" => 1, "bar:jobA" => 2, "baz:jobB" => 3); %h = map { ($job, $setting) = split /:/; $setting => {$job => $x{$_}} } grep { /:/ } keys %x; use Data::Dumper; print(Dumper(\%h))'
```

shows

```
$VAR1 = {
          'jobB' => {
                      'baz' => 3
                    },
          'jobA' => {
                      'bar' => 2
                    }
        };
```

Reply:
* This breaks the deletion.
* This breaks if multiple `:` are present. The splitting should really only occur after the last colon.

But I might be able to make it use `map`.

Reply:
This also breaks if you add `, "bar2:jobA" => 4` (which should result in multiple keys within `jobA`'s nested hash but doesn't.

Reply:
I am sure there is a way to use map to create the hash with correct values even if the same key appears multiple times but if we can't find it then leave it

Reply:
But we're not creating "the hash" here. We're creating one hash containing an arbitrary number of nested hashes - and the nested hashes are not populated at once.